Frontend embed mp4 video

pascal260303 opened the pull request:
The video is now included in the Dataset View and can be watched there.

I had to change the Dataset View to not take everything as URL parameter but to actually fetch data from the backend to access more data.

I also included two simple links to either download the mcap file or open it in foxglove.

resolves #211 

The design of the Dataset View is currently not beautiful but that will be changed in another task.

## How to test
Extract a mp4 video of an mcap file with the `mcap_to_mp4.py` script (or any mp4 file) and put it in the backend/media/ folder (maybe in same subfolder as mcap file).\
Add the video to a mcap file with this:
```python
import os
import django
os.environ.setdefault("DJANGO_SETTINGS_MODULE", "backend.settings")
django.setup()
from restapi.models import File

file = File.objects.get(file="path/to/mcap")
file.video = "path/to/mp4"
file.save()
```
(replace `path/to/mcap` and `path/to/mp4` with the path inside the media folder)

Start the frontend and navigate to the dataset view of the file where you added the video. You should see the video start playing.

DerSimi commented on frontend/app/routes/dataset.tsx:
```diff
@@ -6,6 +7,10 @@ import {
   useLoaderData,
   useSearchParams,
```
```suggestion
```

DerSimi commented on frontend/app/routes/dataset.tsx:
```diff
@@ -1,3 +1,4 @@
+import { Skeleton } from "@mantine/core";
 import { LoaderFunctionArgs } from "@remix-run/node";
 import {
   data,
```
```suggestion
```

DerSimi commented on frontend/app/routes/dataset.tsx:
```diff
@@ -6,6 +7,10 @@ import {
   useLoaderData,
   useSearchParams,
 } from "@remix-run/react";
+import { stringify } from "postcss";
```
```suggestion
```

DerSimi reviewed (changes requested):
Hi Pascal, went through your changes, and everything works nicely as expected. 
![image](https://github.com/user-attachments/assets/c1dc84ec-9f82-4c6a-a3e9-2ddcfc9b4763)
The only issue i found are code smells: Three unused imports. I added the removals as suggestions.

pascal260303 commented:
I removed the unused imports and added a test for the new fetch function. \
I also noticed I forgot the `type` field and added it to the fetch function.
